plugins/camera/camera.h

```diff
@@ -0,0 +1,381 @@
+            float absolute_altitude_m; /**< @brief Altitude AMSL (above mean sea level) in metres */
+            float relative_altitude_m; /**< @brief Altitude relative to takeoff altitude in metres */
+        } position; /**< @brief Position of drone/camera when image was captured. */
+
+        /**
+         * @brief Quaternion type.
+         *
+         * All rotations and axis systems follow the right-hand rule.
+         * The Hamilton quaternion product definition is used.
+         * A zero-rotation quaternion is represented by (1,0,0,0).
+         * The quaternion could also be written as w + xi + yj + zk.
+         *
+         * For more info see: https://en.wikipedia.org/wiki/Quaternion
+         */
+        struct Quaternion {
+            float w; /**< @brief Quaternion entry 0 also denoted as a. */
+            float x; /**< @brief Quaternion entry 1 also denoted as b. */
+            float y; /**< @brief Quaternion entry 2 also denoted as c. */
+            float z; /**< @brief Quaternion entry 3 also denoted as d. */
+        } quaternion; /**< @brief Quaternion of camera orientation. */
+
+        uint64_t time_utc_us; /**< @brief timestamp in UTC (since UNIX epoch) in microseconds. */
+        bool success; /**< @brief True if capture was successful. */
+        int index; /**< @brief Zero-based index of this image since armed. */
+        std::string file_url; /**< URL of image taken to download. */
+    };
+
+    /**
+     * @brief Type for video stream settings.
+     *
+     * Application may call set_video_stream_settings() before starting
+     * video streaming to tell Camera server to use these settings during video streaming.
+     *
+     * @sa set_video_stream_settings(), get_video_stream_info(), start_video_streaming().
+     */
+    struct VideoStreamSettings {
+        float frame_rate_hz = 0.f; /**< @brief Frames per second. */
+        uint16_t resolution_h_pix = 0u; /**< @brief Resolution horizontal in pixels. */
+        uint16_t resolution_v_pix = 0u; /**< @brief Resolution verticle in pixels. */
+        uint32_t bit_rate_b_s = 0u; /**< @brief Bit rate in bits per second. */
+        uint16_t rotation_deg = 0u; /**< @brief Video image rotation clockwise (0-359 degrees) */
+        std::string uri {}; /**< @brief Video stream URI */
+
+        void set_highest()
+        {
+            frame_rate_hz = FRAME_RATE_HIGHEST;
+            resolution_h_pix = RESOLUTION_H_HIGHEST;
+            resolution_v_pix = RESOLUTION_V_HIGHEST;
+            // FIXME: Should bit_rate_b_s be part of set_highest() ?
+            bit_rate_b_s = BIT_RATE_AUTO;
+        }
+
+        constexpr static const float FRAME_RATE_HIGHEST = -1.0f;
+        constexpr static const uint16_t RESOLUTION_H_HIGHEST = UINT16_MAX;
+        constexpr static const uint16_t RESOLUTION_V_HIGHEST = UINT16_MAX;
+        constexpr static const uint32_t BIT_RATE_AUTO = UINT32_MAX;
+    };
+
+    /**
+     * @brief Sets video stream settings.
+     *
```

Comment:
Do we need to give any guidance about what the receiving system will do with invalid information - ie invalid for the specific camera? Does spec state that a -1.0f framerate is treated as "max"?

FYI, if these are setting parameters as defined in the camera definition file, it is probably that this code won't work yet.

Reply:
Yes the spec says `framerate = -1.0f` is treated as "max".

Reply:
>if these are setting parameters as defined in the camera definition file

I'm not sure of this. @julianoes might know.

Reply:
http://mavlink.org/messages/common#SET_VIDEO_STREAM_SETTINGS

Reply:
I don't know the mavlink streaming stuff.

Reply:
What I *know* is that RTSP video stream advertising via MAVLink does not work.  Everything else should work apparently (i.e. video capture and storage just got implemented) but I don't know how much testing has gone on.